File: Srims.WebSite/javascript/srims/papers/PaperIndexedType.js

if (!Srims.papers) 
    Ext.namespace("Srims.papers");

Ext.namespace('Srims.papers.PaperIndexedType');

Srims.papers.PaperIndexedType.SCICD = 'SCICD';
Srims.papers.PaperIndexedType.SCINetWork = 'SCINetWork';
Srims.papers.PaperIndexedType.EICore = 'EICore';
Srims.papers.PaperIndexedType.EINetWork = 'EINetWork';
Srims.papers.PaperIndexedType.ISTP = 'ISTP';
Srims.papers.PaperIndexedType.ISTP = 'SSCI';
Srims.papers.PaperIndexedType.ISTP = 'ISTP_S';

Srims.papers.PaperIndexedType = function(value, metadata){
    switch (value) {
        case 'SCICD':
            return 'SCI光盘';
        case 'SCINetWork':
            return 'SCI网络';
        case 'EICore':
            return 'EI核心';
        case 'EINetWork':
            return 'EI网络';
        case 'SSCI':
            return 'SSCI';
        case 'ISTP':
            return 'ISTP';
        case 'ISTP_S':
            return 'ISTP_S';
        default:
            return '未知';
    }
}
Srims.papers.PaperIndexedTypeFilterItems = [{
    id: 'SCICD',
    text: 'SCI光盘'
}, {
    id: 'SCINetWork',
    text: 'SCI网络'
}, {
    id: 'EICore',
    text: 'EI核心'
}, {
    id: 'EINetWork',
    text: 'EI网络'
}, {
    id: 'ISTP',
    text: 'ISTP'
}, {
    id: 'SSCI',
    text: 'SSCI'
}, {
    id: 'ISTP_S',
    text: 'ISTP_S'
}];
Srims.papers.paperIndexedTypeStore = [['SCICD', 'SCI光盘'], ['SCINetWork', 'SCI网络'], ['EICore', 'EI核心'], ['EINetWork', 'EI网络'], ['ISTP', 'ISTP'],['SSCI', 'SSCI'],['ISTP_S', 'ISTP_S']];

File: Srims.Server.Business/Papers/PaperIndexedType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Srims.Server.Business.Papers
{
    /// <summary>
    /// 论文收录
    /// </summary>
    public enum PaperIndexedType
    {
        /// <summary>
        /// 未知
        /// </summary>
        Unknown = 0,
        /// <summary>
        /// SCI
        /// </summary>
        SCI = 1,
        /// <summary>
        /// SCI光盘
        /// </summary>
        SCICD = 2,
        /// <summary>
        /// SCI网络
        /// </summary>
        SCINetWork = 3,
        /// <summary>
        /// EI
        /// </summary>
        EI = 4,
        /// <summary>
        /// EI核心
        /// </summary>
        EICore = 5,
        /// <summary>
        /// EI网络
        /// </summary>
        EINetWork = 6,
        /// <summary>
        /// ISTP
        /// </summary>
        ISTP = 7,
        /// <summary>
        /// 社会科学引用索引
        /// </summary>
        SSCI = 8,
        /// <summary>
        /// 文科
        /// </summary>
        ISTP_S = 9,
    }
}
